### PR TITLE
STCOR-730: Move useChunkedCQLFetch.js from stripes-erm-components

### PR DESCRIPTION
**Pull request by EthanFreestone**
feat: useChunkedCQLFetch

Added useChunkedCQLFetch hook to stripes-core

STCOR-730

### CODE REVIEW

**Comment by EthanFreestone** [comment]
@zburke @maccabeelevine PR in place with improved useChunkedCQLFetch, can be marked as draft if more discussions are needed before accepting, just wanted to make sure it was more or less ready to go

**Review by zburke — CHANGES_REQUESTED** [review]
This is great code, nicely commented. Thank you! All we need now is test coverage.

File: src/queries/useChunkedCQLFetch.js
Comment (EthanFreestone) [reply src/queries/useChunkedCQLFetch.js:42]: Ah, the stackoverflow I used said the opposite, my first instinct was to use Set. XD Happy to change it though!

File: src/queries/useChunkedCQLFetch.js
Comment (maccabeelevine) [inline src/queries/useChunkedCQLFetch.js:22]: When testing this hook please include one test case with > 300 *unique* IDs (60 * 5).  I could not yet get that scenario to work in ui-inventory as the very last batch of 0<x<=60 IDs would consistently fail, but I was on vacation last week and haven't yet figured out why.

Comment (EthanFreestone) [reply src/queries/useChunkedCQLFetch.js:27]: Have written basic test coverage, including a test case for that, but since this is unit testing, it's possible something is going wrong with the endpoint returns etc?

Comment (maccabeelevine) [reply src/queries/useChunkedCQLFetch.js:27]: Certainly possible yes.

**Review by zburke — APPROVED** [review]
LGTM

File: src/queries/useChunkedCQLFetch.js
Comment (EthanFreestone) [reply src/queries/useChunkedCQLFetch.js:27]: @maccabeelevine Have you managed to get this working as written yet or is it still not fetching the final page?

Comment (maccabeelevine) [reply src/queries/useChunkedCQLFetch.js:27]: @EthanFreestone Sorry no, have not had time yet.  It's a very artificial scenario now that the de-duping is in place in ui-inventory so I will have to come up with some more creative test data.  No timeline but I hope soon.

Comment (EthanFreestone) [reply src/queries/useChunkedCQLFetch.js:27]: @maccabeelevine Apologies for pushing again, but I'd like to merge this whenever possible. Currently not seeing any problems with the equivalent code in `stripes-erm-components`, and would like to merge this if possible.

Could you please try to recreate your issue with it?

Comment (maccabeelevine) [reply src/queries/useChunkedCQLFetch.js:27]: @EthanFreestone I think you are good to merge it, as you said if there actually is an issue here it is almost definitely on the endpoint side, and the changes made there will have mitigated any real-world example.  Apologies I haven't been able to reproduce yet and won't for a while more, I would not hold up the merge.

Comment (EthanFreestone) [reply src/queries/useChunkedCQLFetch.js:27]: Thank you Maccabee :) I'll merge asap, hopefully the issue doesn't rear it's head for you down the line.

**Pull request merged**
